std/range/package.d

```diff
@@ -4173,10 +4195,10 @@ private string lockstepMixin(Ranges...)(bool withIndex)
             }
             return res;
         }
-    }, params.join(", "), withIndex ? "size_t index = 0;" : "",
+    }, name, params.join(", "), indexDef,
        emptyChecks.join(" && "), dgArgs.join(", "),
        popFronts.join("\n                "),
-       withIndex ? "index++;" : "");
```

Comment:
You're commenting on a line that's removed. The indexing when using lockstep with foreach_reverse will be just like when using foreach_reverse with an array.

The point about differing lengths is a pretty good one, though. If people really want some kinda index, they could just lockstep(iota(r1.length-1, -1, -1), r1, r2).